weblate/formats/tests/xwiki/test_xwiki.py

```diff
@@ -0,0 +1,468 @@
+import os
+from xml.etree import ElementTree
+
+from django.utils.encoding import force_text
+
+from weblate.formats.tests.test_formats import AutoFormatTest
+from weblate.formats.xwiki import (
+    XWikiFullPageFormat,
+    XWikiPagePropertiesFormat,
+    XWikiPropertiesFormat,
+)
+from weblate.lang.models import Language
+
+XWIKI_TEST_DATA = os.path.join(
+    os.path.dirname(os.path.abspath(__file__)),
+    'data'
+)
+
+
+def get_xwiki_test_file(name):
+    """Return filename of test file."""
+    return os.path.join(XWIKI_TEST_DATA, name)
+
+
+XWIKI_PROPERTIES = get_xwiki_test_file('xwiki.properties')
+XWIKI_PAGE_PROPERTIES = get_xwiki_test_file("XWikiSource.xml")
+XWIKI_FULL_PAGE = get_xwiki_test_file("XWikiFullPage.xml")
+
+
+class XWikiPropertiesFormatTest(AutoFormatTest):
+    FORMAT = XWikiPropertiesFormat
+    FILE = XWIKI_PROPERTIES
+    BASE = ''
+    MIME = 'text/plain'
+    COUNT = 10
+    COUNT_CONTENT = 8
+    EXT = 'properties'
+    MASK = 'java/xwiki_*.properties'
+    EXPECTED_PATH = 'java/xwiki_cs-CZ.properties'
+    FIND = 'job.question.button.confirm'
+    FIND_CONTEXT = 'job.question.button.confirm'
+    FIND_MATCH = 'Confirm the operation {0}'
+    MATCH = '\n'
+    NEW_UNIT_MATCH = b'\nkey=Source string\n'
+    EXPECTED_FLAGS = ''
+
+    def assert_same(self, newdata, testdata):
+        self.assertEqual(
+            force_text(newdata).strip().splitlines(),
+            force_text(testdata).strip().splitlines(),
+        )
```

Comment:
Again, you could inherit from PropertiesFormatTest to get this behavior instead of copying it.